### PR TITLE
MAINT make `Scattering2D` compatible with TF>=2.5

### PR DESCRIPTION
**Pull request by lostanlen**
Addresses #775 for 2D

### CODE REVIEW

File: kymatio/scattering2d/filter_bank.py [hunk 1]
```diff
@@ -30,30 +30,25 @@ def filter_bank(M, N, J, L=8):
...
     phi_signal_fourier = fft2(phi_signal)
     # drop the imaginary part, it is zero anyway
-    phi_signal_fourier = np.real(phi_signal_fourier)
-    filters['phi']['j'] = J
+    filters['phi'] = {'levels': [], 'j': J}
     for res in range(J):
-        phi_signal_fourier_res = periodize_filter_fft(phi_signal_fourier, res)
-        filters['phi'][res] = phi_signal_fourier_res
+        filters['phi']['levels'].append(
+            periodize_filter_fft(filters['phi']['levels'][0], res))
```
Comment (MuawizChaudhary) [inline kymatio/scattering2d/filter_bank.py:51]: This is missing the `phi_sginal_fourier`

**Review by MuawizChaudhary — COMMENTED** [review]
this should be correct.

File: kymatio/scattering2d/filter_bank.py [hunk 2]
```diff
@@ -30,30 +30,25 @@ def filter_bank(M, N, J, L=8):
...
             psi_signal_fourier = fft2(psi_signal)
             # drop the imaginary part, it is zero anyway
-            psi_signal_fourier = np.real(psi_signal_fourier)
+            psi_levels = []
             for res in range(min(j + 1, max(J - 1, 1))):
-                psi_signal_fourier_res = periodize_filter_fft(
-                    psi_signal_fourier, res)
-                psi[res] = psi_signal_fourier_res
+                psi_levels.append(periodize_filter_fft(psi_levels[0], res))
+            psi['levels'] = psi_levels
```
Comment (MuawizChaudhary) [inline kymatio/scattering2d/filter_bank.py:42]: Missing the element at 0

**Comment by lostanlen** [comment]
thanks for your help, @MuawizChaudhary . bugfixed

**Review by MuawizChaudhary — APPROVED** [review]
Looks good to me. I'll wait for the tests to finish runnin.

**Comment by MuawizChaudhary** [comment]
I'll try to get to it tonight.

**Comment by lostanlen** [comment]
what's happening with Jenkins?

**Comment by MuawizChaudhary** [comment]
@janden  I created PR #888 resolving the lack of tests. They fail prior to the merges.

**Comment by MuawizChaudhary** [comment]
This PR goes a small way of resolving issue #817 , as seen in [this notebook, the keras example fails](https://colab.research.google.com/drive/1_G_vFlAemNtJregIWdA6T8H3B6BUoV1N?usp=sharing), but [here](https://colab.research.google.com/drive/1PdazlZMUkY4Lfh1Zv5HX2R3Hh2WH262X?usp=sharing) on @lostanlen's fix-775-2D branch it works! 


The tests in #888 also ensure correctness.

File: kymatio/scattering2d/frontend/torch_frontend.py [hunk 3]
```diff
@@ -33,19 +33,13 @@ def register_filters(self):
 
         n = 0
 
-        for c, phi in self.phi.items():
-            if not isinstance(c, int):
-                continue
-
-            self.register_single_filter(phi, n)
+        for phi_level in self.phi['levels']:
```
Comment (janden) [inline kymatio/scattering2d/frontend/torch_frontend.py:36]: Much cleaner.

**Review by janden — APPROVED** [review]
So what about 1D?

File: kymatio/scattering2d/frontend/torch_frontend.py (hunk 3 above)
Comment (lostanlen) [reply kymatio/scattering2d/frontend/torch_frontend.py:36]: à quelque chose malheur est bon ! ;)

**Comment by janden** [comment]
> @janden I merged 1d in #882

Got it.

**Pull request merged**